src/test/java/perosnal/spotifymatcher/service/UserServiceTest.java

```diff
@@ -0,0 +1,131 @@
+package perosnal.spotifymatcher.service;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import perosnal.spotifymatcher.model.User;
+import perosnal.spotifymatcher.repository.UserRepository;
+import perosnal.spotifymatcher.util.AuthorizedActionResult;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.lenient;
+
+@ExtendWith(MockitoExtension.class)
+class UserServiceTest {
+
+    @Mock
+    private UserRepository userRepository;
+    @Mock
+    private SpotifyApiService spotifyApiService;
+    @InjectMocks
+    private UserService userService;
```

Comment:
[Avoid `@InjectMocks` to prevent silent errors](https://tedvinke.wordpress.com/2014/02/13/mockito-why-you-should-not-use-injectmocks-annotation-to-autowire-fields/). Object under test should be initialized with a constructor.

```suggestion
class UserServiceTest {
    
    private final UserRepository userRepository = Mockito.mock(UserRepository.class);
    private final SpotifyApiService spotifyApiService = Mockito.mock(SpotifyApiService.class);
    
    private final UserService userService = new UserService(userRepository, spotifyApiService);
```

ᴛʜɪs ᴀᴄᴛɪᴏɴ ᴡᴀs ᴘᴀʀᴛɪᴀʟʟʏ ᴀᴜᴛᴏᴍᴀᴛᴇᴅ ʙʏ [GɪᴛHᴜʙ's ʀᴇsᴘᴏɴsᴇ ᴛᴇᴍᴘʟᴀᴛᴇs](https://github.com/settings/replies)